plugins/versionpress/src/Database/schema-readme.md

```diff
@@ -1,124 +1,244 @@
+
+
+```
+option:
+  value-references:
+    option_name@option_value:
+      page_on_front: post
+      default_category: term
+      ...
+```
+
+If the entity type needs to be determined dynamically it can reference a PHP function:
+
+```
+postmeta:
+  value-references:
+    meta_key@meta_value:
+      _menu_item_object_id: '@\VersionPress\Database\VpidRepository::getMenuReference'
+```
+
+Note that there are no parenthesis at the end of this (it's a method reference, not a call) and that it is prefixed with `@`. The function gets the entity as a parameter and returns a target entity name. For example, for `_menu_item_object_id`, the function looks for a related DB row with `_menu_item_type` and returns its value.
  
- The field `parent-reference` contains the name of one of the simple references. This specifies within which entity will be the child entity saved.
+If the ID is in a serialized object, you can specify the path by a suffix of the source column. It looks like array access but also supports regular expressions, for example:
+
+```
+option:
+  value-references:
+    option_name@option_value:
+      widget_pages[/\d+/]["exclude"]: post
+```
+
+To visualize this, the `widget_pages` option contains a value like `a:2:{i:2;a:3:{s:5:"title";s:0:"";s:7:"exclude";s:7:"1, 2, 3";...}...}` which, unserialized, looks like this:
+
+```
+[
+  2 => [
+    "title" => "",
+    "sortby" => "post_title",
+    "exclude" => "1, 2, 3"
+  ],
+  "_multiwidget" => 1
+]
+```
+
+The schema says that the numbers in the "exclude" key point to posts.
+
+Value references also support wildcards in the name of the source column. It's useful e.g. for options named `theme_mods_<name of theme>`. An example that mixes this with the serialized data syntax is:
+
+```
+option:
+  value-references:
+    option_name@option_value:
+      theme_mods_*["nav_menu_locations"][/.*/]: term
+      theme_mods_*["header_image_data"]["attachment_id"]: post
+      theme_mods_*["custom_logo"]: post
+```
+
+
+### M:N references
+
+Some entities are in an M:N relationship like posts and term_taxonomies. The format to capture this is:
+
+```
+mn-references:
+  <junction_table_name>.<column_name>: <foreign_entity_name>
+```
+
+This is a concrete example from the post entity:
+
+```
+post:
+  mn-references:
+    term_relationships.term_taxonomy_id: term_taxonomy
+```
+
+One entity is considered "master" which is kind of arbitrary (technically, they are equal) but here, we decided that posts will store tags and categories, not the other way around. INI files of posts will store the references.
+
+References can also be prefixed with a tilde (`~`) which makes them virtual:
+
+```
+mn-references:
+  ~<junction_table_name>.<column_name>: <foreign_entity_name>
+```
+
+Virtual references are not stored in INI files but the relationships are checked during reverts. For example, when restoring a post that references a category that no longer exists, this operation will fail.
```

Comment:
This is an example of standard M:N reference. A virtual reference works, for example, in a situation, when a revert would delete a category (revert of `term_taxonomy/create`) and there is some post referencing it.